### PR TITLE
Fix all psalm issues in theming app

### PR DESCRIPTION
**Pull request by CarlSchwan**
This also makes composer psam:update-baseline not silently ignore new errors.

image
:)

### CODE REVIEW

File: apps/theming/lib/Controller/IconController.php [hunk 1]
```diff
@@ -111,7 +107,8 @@ public function getThemedIcon(string $app, string $image): Response {
 	 * @NoCSRFRequired
 	 *
 	 * @param $app string app name
-	 * @return FileDisplayResponse|DataDisplayResponse
+	 * @psalm-return FileDisplayResponse|DataDisplayResponse
+	 * @return Response
```
Comment (come-nc) [inline apps/theming/lib/Controller/IconController.php:111]: Why is this change needed? Which tool does not support Union types in @return?
They are available since PHP 8.0

File: apps/theming/lib/ThemingDefaults.php [hunk 2]
```diff
@@ -157,6 +157,10 @@ public function getBaseUrl() {
 		return $this->config->getAppValue('theming', 'url', $this->url);
 	}
 
+	/**
+	 * @psalm-suppress InvalidReturnStatement
+	 * @psalm-suppress InvalidReturnType
```
Comment (come-nc) [inline apps/theming/lib/ThemingDefaults.php:162]: suppress directives should be explained by a comment in my opinion.

File: apps/theming/lib/ThemingDefaults.php [hunk 3]
```diff
@@ -404,8 +408,8 @@ protected function getCustomFavicon(): ?ISimpleFile {
 	 * Increases the cache buster key
 	 */
 	private function increaseCacheBuster() {
```
Comment (come-nc) [inline apps/theming/lib/ThemingDefaults.php:410]:
```suggestion
	private function increaseCacheBuster(): void {
```

**Review by come-nc — COMMENTED** [review]
Awesome work!

File: apps/theming/lib/ThemingDefaults.php [hunk 4]
```diff
@@ -157,6 +157,11 @@ public function getBaseUrl() {
 		return $this->config->getAppValue('theming', 'url', $this->url);
 	}
 
+	/**
+	 * We pass a string and sanitizeHTML will return a string to in that case
```
Comment (come-nc) [inline apps/theming/lib/ThemingDefaults.php:161]:
```suggestion
	 * We pass a string and sanitizeHTML will return a string too in that case
```

**Review by come-nc — APPROVED** [review]

**Review by PVince81 — APPROVED** [review]
:+1:
